### PR TITLE
Allow arm64 jit when memory base is not nicely aligned

### PR DESCRIPTION
**Pull request by unknownbrackets**
This basically adds the arm32-style ML_ARMREG_AS_PTR to arm64, but only uses it when pointerify is disabled.

This at least improves the situation on 64-bit iOS 10.  Only really tested with Final Fantasy 4.  It may crash in iOS 11 or with jailbreaks or etc., and I'm not planning to do anything about that right now.  Don't ask.

~~FlushIcache() seems to crash a lot, though, not sure why.~~ It was because it didn't account for the start offset.

FWIW: In the singular game I tested, it seems like 64-bit is slower by about 10% (550% -> 500%.)  This may be the jit, since IR seemed roughly the same (around 250% with fast forward.)

-[Unknown]

### CODE REVIEW

**Comment by iOS4all** [comment]
Thanks for taking care about iOS. We really appreciate a lot.
But we have an issue.

This is the last update on Cydia
img_0610

Even though inside ppsspp it's still on v1.4.2-575

**Comment by unknownbrackets** [comment]
That's not at all related to this PR.  I'm not planning on spending any time looking at Cydia.

-[Unknown]

**Comment by iOS4all** [comment]
Thank you so much bro.
I'll waiting for someone to update ppsspp on Cydia.

**Comment by hrydgard** [comment]
Very nice, thanks. @angelXwind , any chance you can have a look at the cydia buildbot?

Regarding 32 vs 64-bit, there are a few sections of the JIT that are still disabled in 64-bit (lwl/lwr etc). Plus, 32-bit is removed in ios 11 and will be deprecated on Android 2019 so it's not like we have a choice long term anyway :)

**Comment by unknownbrackets** [comment]
Well, keep in mind I've disabled two of the nicer optimizations in the 64-bit jit - I didn't say it's slower on Android.  Plus, agreed, there's more room to optimize.  Just trying to temper expectations.

-[Unknown]

File: Core/MIPS/ARM64/Arm64CompFPU.cpp [hunk 1]
```diff
@@ -111,7 +111,11 @@ void Arm64Jit::Comp_FPULS(MIPSOpcode op)
 			} else {
 				skips = SetScratch1ForSafeAddress(rs, offset, SCRATCH2);
 			}
-			MOVK(SCRATCH1_64, ((uint64_t)Memory::base) >> 32, SHIFT_32);
+			if (jo.enablePointerify) {
+				MOVK(SCRATCH1_64, ((uint64_t)Memory::base) >> 32, SHIFT_32);
+			} else {
+				ADD(SCRATCH1_64, SCRATCH1_64, MEMBASEREG);
```
Comment (unknownbrackets) [inline Core/MIPS/ARM64/Arm64CompFPU.cpp]: Note: Would be better to just use MEMBASEREG as the last arg to LDR and not do either of these?

-[Unknown]

Comment (hrydgard) [reply Core/MIPS/ARM64/Arm64CompFPU.cpp]: Well, that form of LDR takes an immediate offset, not a register, but changing to the register sum LDR would indeed probably be a win. Definitely not a loss. Don't remember, do we initialize MEMBASEREG even if pointerization is enabled?

Comment (hrydgard) [reply Core/MIPS/ARM64/Arm64CompFPU.cpp]: Though I suppose not needing to reload registers to "depointerize" should still be quite a win.

**Pull request merged**